### DISCUSSION TITLE
integrating with headless selenium

### DISCUSSION THREAD

**Discussion by samarthgupta437** [hyperjumptech/monika#537] [General]
Our product website integrates login authentication via auth-0.  It has not been possible to automate login via simple http post request sequence and requires actually going via a browser. 
We have been working by writing selenium scripts for login and using the jwt token obtained in response. However test start to fail once jwt expires. Is there a way to :

1. invoke selenium / python / shell scripts as part of probe ?
2. if auth / login fails a way to call above to get new token and see if that was the issue ?

**Comment by nicnocquee** [comment]
Hello @samarthgupta437 ,

1. It is not possible to run a script as part of probe at this moment.
2. If I understood your use case correctly, you probably can use the chaining request feature. Check out this article https://medium.com/hyperjump-tech/be-alerted-when-your-authentication-api-is-slow-with-monika-a-guide-for-chaining-request-a63801df8b39
